### PR TITLE
Add an indicative limit for judging if a PR is too big in PR guidelines

### PR DESCRIPTION
**Pull request by pchila**
@ycombinator: Let's wait till at least 2/3 of the reviewers tagged on this PR have approved it.  It's an unusually high bar but I'd rather we clear that first than have pushback later.

## What does this PR do?

Add indications about expected size of PRs in CONTRIBUTING.md

## Why is it important?

To improve code review experience.

## Checklist

- ~~[ ] My code follows the style guidelines of this project~~
- ~~[ ] I have commented my code, particularly in hard-to-understand areas~~
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- ~~[ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)~~
- ~~[ ] I have added an integration test or an E2E test~~

## Related issues

-

## Questions to ask yourself

- How are we going to support this in production?
- How are we going to measure its adoption?
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

**Review by swiatekm — APPROVED** [review]

**Review by pkoutsovasilis — APPROVED** [review]

**Comment by pierrehilbert** [comment]
I'm also adding folks from my team as we are working also on the Agent repository.

**Review by mauri870 — APPROVED** [review]

**Review by leehinman — APPROVED** [review]

**Review by cmacknz — APPROVED** [review]

**Review by ycombinator — APPROVED** [review]

**Review by faec — APPROVED** [review]

**Review by michel-laterman — APPROVED** [review]

**Review by michalpristas — APPROVED** [review]
500 seems like a reasonable limit

**Review by VihasMakwana — APPROVED** [review]

**Review by blakerouse — COMMENTED** [review]
Does this count unit tests in this line count? That seems low if it included unit tests, some changes have way more tests than line changes and I can see that easily going over 500 lines.

**Review by AndersonQ — APPROVED** [review]

**Comment by pchila** [comment]
> Does this count unit tests in this line count? That seems low if it included unit tests, some changes have way more tests than line changes and I can see that easily going over 500 lines.

That is why I added the clause 

> (excluding generated code or other changes that normally should not require close review by a human)

to catch stuff like testdata in a unit test, generated code or anything else that does not require the same effort for review as code.

For example a super complicated unit test of 2k lines of code for a single testcase may be deemed too complicated and too big (In that case the PR could be split by separating the tests  and the feature code in 2 PRs  maybe) but in general single unit tests should be short and easy to understand

The guideline is there to remind people that we should strive to keep the PR size small and that is ok to ask for a big change to be split in smaller ones and as @cmacknz said it's not a mindlessly automated hard limit... it will be evaluated case by case

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -44,6 +44,8 @@ Keep these in mind as both authors and reviewers of PRs:
   - The PR must contain related changes and leave out unrelated changes (e.g. refactorings, etc. that could be their own PR instead).
   - If the PR could be broken up into two or more PRs either "vertically" (by separating concerns logically) or horizontally (by sharding the PR into a series of PRs --- usually works well with mass refactoring or cleanup type PRs), it should. A set of such related PRs can be tracked and given context in a meta issue.
 
+* The PR changeset should be rather small: 500 lines of code (excluding generated code or other changes that normally should not require close review by a human) is the soft limit that should be used to judge if a PR is "too big".
```
Comment (andrzej-stencel) [inline CONTRIBUTING.md:47]: LGTM :+1: 500 lines is also used in [OTel Collector guidelines](https://github.com/open-telemetry/opentelemetry-collector/blob/v0.114.0/CONTRIBUTING.md#how-to-structure-prs-to-get-expedient-reviews):

> We recommend that any PR (unless it is trivial) to be smaller than 500 lines (excluding go mod/sum changes) in order to help reviewers to do a thorough and reasonably fast reviews.

**Review by khushijain21 — APPROVED** [review]

**Review by blakerouse — APPROVED** [review]
Thanks for the clarity. +1

**Comment by pierrehilbert** [comment]
@kaanyalti I think we are only missing your approval here

**Review by kaanyalti — APPROVED** [review]

**Comment by ycombinator** [comment]
buildkite test this

**Pull request merged**